src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -240,7 +240,11 @@ internal static IntPtr NativeDllHandler(Assembly assembly, string libraryName)
         {
             s_nativeDllSubFolder ??= GetNativeDllSubFolderName(out s_nativeDllExtension);
             string folder = Path.GetDirectoryName(assembly.Location);
-            string fullName = Path.Combine(folder, s_nativeDllSubFolder, libraryName) + s_nativeDllExtension;
+            string fullName = Path.Combine(folder, s_nativeDllSubFolder, libraryName);
+            if (!fullName.EndsWith(s_nativeDllExtension, StringComparison.OrdinalIgnoreCase))
+            {
+                fullName += s_nativeDllExtension;
```

Comment:
Should we overwrite the extension if developer point it _explicitly_?

Reply:
Will this break existing PSGallery modules? In the SQLite example, none of the project code specifies a DLL name, it is all done by the DLL linking references in the binaries. The native DLLs are all referenced by the existing binary "System.Data.SQLite.dll" which was simply downloaded from the official binaries. There should be no need for PowerShell to append an additional .so, .dylib or dll as all platforms will happily load shared libraries with a .dll extension. The platform directory ( eg osx-arm64 or linux-arm ) has already fully qualified it in terms of architecture.

Reply:
I mean we shouldn`t add any extension if developer explicitly set an extension in `DllImport` like `filename.qqq` - we should exactly use the name.

Reply:
@rhubarb-geek-nz The code is for _native_ dll name. Before the fix for "interop.dll" we get unexpected "interop.dll.dll" or "interop.dll.so".